kratos/input_output/vtk_output.cpp

```diff
@@ -0,0 +1,452 @@
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4)
+             cell_type = 9; // quad 2d
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4)
+             cell_type = 10; // tetra 3d
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line2D2)
+             cell_type = 3; // line 2d or 3d
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line3D2)
+             cell_type = 3; // line 2d or 3d
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point2D)
+             cell_type = 1;
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point3D)
+             cell_type = 1;
+        if( cell_type < 0)
+            KRATOS_ERROR<<"Modelpart contains elements with geometries for which no VTK-output is implemented!"<<std::endl;
+
+        WriteScalarDataToFile( (int)cell_type, rFileStream);
+        if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
+    }
+
+    // write conditions types
+    for (const auto& condition : local_mesh.Conditions())
+    {
+        int  cell_type = -1;
+
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle2D3)
+             cell_type = 5; // triangle 2d
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4)
+             cell_type = 9; // quad 2d
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4)
+             cell_type = 10; // tetra 3d
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line2D2)
+             cell_type = 3; // line 2d or 3d
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line3D2)
+             cell_type = 3; // line 2d or 3d
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point2D)
+             cell_type = 1;
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point3D)
+             cell_type = 1;
+        if( cell_type < 0)
+            KRATOS_ERROR<<"Modelpart contains elements with geometries for which no VTK-output is implemented!"<<std::endl;
+
+        WriteScalarDataToFile( (int)cell_type, rFileStream);
+        if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
+    }
+
+
+}
+
+void VtkOutput::WriteNodalResultsAsPointData(const ModelPart &rModelPart, std::ofstream& rFileStream)
+{
+    const auto& local_mesh = rModelPart.GetCommunicator().LocalMesh();
+    // write nodal results header
+    Parameters nodal_results = this->mrOutputSettings["nodal_solution_step_data_variables"];
```

Comment:
One way would be to extract the variables into an std::vector and then loop it, this way you don't have the repeated access to Paramters and Components